### DISCUSSION TITLE
typescript + createStore + produce

### DISCUSSION THREAD

**Discussion by asafs94** [solidjs/solid#692] [Q&A]
Hey, let me start by saying that I really like SolidJs :)
I tried to create a store while using typescript and I have an issue with inference.
When trying to use `setStore` with produce. The first argument of the callback I pass to produce is treated as unknown by typescript.
I wanted to know if there is any better way to deal with that other than specifying the parameter type explicitly every time I use `setStore`.

This is my code: 
```
import { createStore, produce } from "solid-js/store";
import { AppCanvas, WithoutId } from "@/entities";
import { createId } from "@/utils";

type CanvasesState = {
  ids: AppCanvas["id"][];
  entities: Record<AppCanvas["id"], AppCanvas>;
};

const [canvases, setCanvases] = createStore({
  ids: [],
  entities: {},
} as CanvasesState);

export const addCanvas = (canvas: WithoutId<AppCanvas>) => {
  const id = createId();
  setCanvases(
    produce((state /** This is treated as unknown */) => {
      state.ids.push(id);
      state.entities[id] = { id, ...canvas };
    })
  );
};

```

**Comment by ryansolid** [comment]
There might be something we can do with types.. but not really my area of specialty. The `setStore` types are very complicated. I think produce takes a generic but that's still you manually typing it. Not sure if there is a way to use inference here.
